File: Git-Templates/ProjectLanguageFiles/SNU/2D/ProgrammingTools/!/IDE/Mercury/Mercury/PROJECT_LANG_1_V1.m
% Start of script
:- module project-language-file-one.
:- interface.
:- import_module io.
:- pred main(io::di, io::uo) is det.

:- implementation.
main(!IO) :-
	io.write_string("Project language file 1\nFor: SNU/2D/ProgrammingTools/IDE/Mercury\nAbout:\nI decided to make Mercury the main project language file for this project (SNU / 2D / Programming Tools / IDE / Mercury) as this is a Mercury IDE, and it needs its main language to be represented here.\n", !IO).

% Output:

% Project language file 1
% For: SNU/2D/ProgrammingTools/IDE/Mercury
% About:
% I decided to make Mercury the main project language file for this project (SNU / 2D / Programming Tools / IDE / Mercury) as this is a Mercury IDE, and it needs its main language to be represented here.

% File info
% File type: Mercury source file (*.m)
% File version: 1 (2022, Thursday, September 15th at 6:35 pm PST)
% Line count (including blank lines and compiler line): 24

% End of script

File: Git-Templates/ProjectLanguageFiles/SNU/2D/ProgrammingTools/!/IDE/ML/ML/PROJECT_LANG_1_V1.ml
(* Start of script *)
structure Rational : ARITH =
struct
        datatype t = Rat of int * int
        val zero = Rat (0, 1)
        fun succ (Rat (a, b)) = Rat (a + b, b)
        fun sum (Rat (a, b), Rat (c, d)) = Rat (a * d + c * b , b * d)
end
(* Project language file 1*)
(* For: SNU/2D/ProgrammingTools/IDE/ML/ *)
(* About:*)
(* I chose ML as the first project language for this project (SNU/2D/ProgrammingTools/IDE/ML) as this is a ML IDE, and should be represented with the ML programming language. *)

(* File info *)
(* File type: Standard ML source file (.ml)*)
(* File version: 1 (2022, Sunday, February 13th at 4:27 pm)* )
(* Line count (Including blank lines and compiler line): 20 *)

(* End of script *)
